doc/source/howtos/encrypt-traffic.rst

```diff
@@ -0,0 +1,307 @@
+
+xtrabackup
+----------
+
+This is the default SST method,
+which uses Percona XtraBackup to perform non-blocking transfer of files.
+For more information,
+see :ref:`xtrabackup_sst`.
+
+Encryption mode for this method is selected using the :option:`encrypt` option.
+Depending on the mode you select, other options will be required.
+
+* To enable built-in XtraBackup encryption, use the following configuration:
+
+  .. code-block:: text
+
+     [sst]
+     encrypt=1
+     encrypt-algo=AES256
+     encrypt-key=A1EDC73815467C083B0869508406637E
+
+  In this example, you can set ``encrypt-key-file`` instead of ``encrypt-key``.
+
+  For more information,
+  see `Encrypted Backups <https://www.percona.com/doc/percona-xtrabackup/2.4/innobackupex/encrypted_backups_innobackupex.html>`_.
+
+* To enable SST encryption based on OpenSSL
+  with the certificate authority (``tca``) and certificate (``tcert``) files:
+
+  .. code-block:: text
+
+     [sst]
+     encrypt=2
+     tcert=/path/to/server.pem
+     tca=/path/to/server.crt
+
+  For more information,
+  see `Securing Traffic Between two Socat Instances Using SSL <http://www.dest-unreach.org/socat/doc/socat-openssltunnel.html>`.
+
+* To enable SST encryption based on OpenSSL
+  with the key (``tkey``) and certificate (``tcert``) files:
+
+  .. code-block:: text
+
+     [sst]
+     encrypt=3
+     tcert=/path/to/server.pem
+     tkey=/path/to/server.key
+
+* To enable SST encryption based on SSL files generated by MySQL,
+  use the following configuration::
+
+   [sst]
+   encrypt=4
+   ssl-ca=ca.pem
+   ssl-cert=server-cert.pem
+   ssl-key=server-key.pem
+
+  For more information, see
+  http://dev.mysql.com/doc/refman/5.7/en/creating-ssl-files-using-openssl.html
+
+  The following procedure shows how to generate the :file:`ca.pem`,
+  :file:`server-cert.pem`, and :file:`server-key.pem` files.
+
+  1. To generate the CA certificate::
+
+      openssl genrsa 2048 > ca-key.pem
+      openssl req -new -x509 -nodes -days 3600 -key ca-key.pem -out ca.pem
+
+  #. To generate the server certificate, remove passphrase, and sign it::
+
+      openssl req -newkey rsa:2048 -days 3600 -nodes -keyout server-key.pem -out server-req.pem
+      openssl rsa -in server-key.pem -out server-key.pem
+      openssl x509 -req -in server-req.pem -days 3600 -CA ca.pem -CAkey ca-key.pem -set_serial -1 -out server-cert.pem
+
+  #. (Optional) To generate the client certificate,
+     remove passphrase, and sign it::
+
+      openssl req -newkey rsa:2048 -days 3600 -nodes -keyout client-key.pem -out client-req.pem
+      openssl rsa -in client-key.pem -out client-key.pem
+      openssl x509 -req -in client-req.pem -days 3600 -CA ca.pem -CAkey ca-key.pem -set_serial 01 -out client-cert.pem
+
+  There are two ways to deploy SSL files across the cluster:
+
+  * Use the same files for all machines in the configuration.
```

Comment:
Alexey again wherever there are multiple option clearly indicate this by tagging
ALTERNATIVE-1:
ALTERNATIVE-2: